### PR TITLE
fix: remove dist files from docker templates

### PR DESCRIPTION
**Pull request by Gaitholabi**
The goal of this PR is to make rolling updates a much simpler process.  This is achieved by removing the `.dist` functionality and documenting a separate process to add customizations to the deployment.  Documentation will be added on how to migrate or add customizations.

### CODE REVIEW

File: templates/docker-compose/Makefile [hunk 1]
```diff
@@ -4,6 +4,7 @@ include .env
 
 DOCKER_BIN=docker
 DOCKER_COMPOSE_BIN=docker-compose
+GIT_BIN=git
 COMPOSE_FILES=-f docker-compose.yml -f docker-compose.prod.yml
```
Comment (bytestream) [inline templates/docker-compose/Makefile:8]: What happens if you need to add `-f docker-compose.override.yml`?

Comment (Gaitholabi) [reply templates/docker-compose/Makefile:8]: should just recommend in docs that they do that on a separate file that is not tracked in repo

File: templates/docker-compose/.env [hunk 2]
```diff
@@ -33,7 +33,6 @@ HOST=example.com
 # Misc
 APP_VERSION=3.6.2
 MAILTO=example@supportpal.com
-DOMAIN_NAME=example.com
```
Comment (bytestream) [inline templates/docker-compose/.env:36]: Not sure about renaming `.env.dist` to `.env`. Some stuff like `MAILTO` does need to be changed by the user?

File: templates/docker-monolithic/upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    exit 1
+    ;;
+  esac
+  shift
+done
+
+
+upgrade() {
+    docker-compose down -v
+    curl -fLsS https://raw.githubusercontent.com/supportpal/helpdesk-install/master/templates/docker-monolithic/docker-compose.yml -o docker-compose.yml
```
Comment (bytestream) [inline templates/docker-monolithic/upgrade.sh:26]: this should probably backup the original file before it overwrites it?

File: templates/docker-compose/Makefile.dist [hunk 4]
```diff
@@ -13,7 +14,7 @@ help:
 
 .PHONY: install
 install: configure create_secrets create_volumes ## Install the helpdesk.
-	$(DOCKER_COMPOSE_BIN) $(COMPOSE_FILES) up -d gateway supportpal db redis
+	$(DOCKER_COMPOSE_BIN) $(COMPOSE_FILES) up -d supportpal
```
Comment (bytestream) [inline templates/docker-compose/Makefile.dist:17]: How does this work? Those services need to be up?

File: templates/docker-compose/setup.sh [hunk 5]
```diff
@@ -20,13 +20,15 @@ skip_clone=0
...
 email=
+mode="install"
 
 while [[ "$#" -gt 0 ]]; do
   case $1 in
   -h|--help) echo "$usage" ; exit 0 ;;
   -n) interactive=0 ;;
   -H|--host) host="$2" ; shift ;;
   -e|--email) email="$2" ; shift ;;
+  -m|--migrate) mode="configure" ;;
```
Comment (bytestream) [inline templates/docker-compose/setup.sh:31]: `-t|--target` and `target=` might be a better name

File: templates/docker-compose/docker-compose.yml [hunk 6]
```diff
@@ -31,7 +31,6 @@ services:
         environment:
             REDIS_HOST: ${CACHE_SERVICE_NAME}
             REDIS_PORT: ${REDIS_PORT}
-            HOST: ${HOST}
```
Comment (bytestream) [inline templates/docker-compose/docker-compose.yml:34]: The `HOST` environment var doesn't appear to be used anywhere now?

File: templates/docker-compose/Makefile.dist (hunk 4 above)
Comment (Gaitholabi) [reply templates/docker-compose/Makefile.dist:17]: At this point, we only want to kick off the command line installation, we only need the main web service, and docker-compose will also up the dependencies based on `depends_on`

File: templates/docker-monolithic/setup.sh [hunk 7]
```diff
@@ -173,14 +173,6 @@ configure() {
   fi
   curl -fLsS https://raw.githubusercontent.com/supportpal/helpdesk-install/master/templates/docker-monolithic/docker-compose.yml -o docker-compose.yml
 
-  # guess the hostname
```
[the remark this answers is not in the document]
Comment (Gaitholabi) [reply templates/docker-monolithic/setup.sh:176]: okay, could add it to a compose-override then

File: templates/docker-compose/docker-compose.yml (hunk 6 above)
Comment (bytestream) [reply templates/docker-compose/docker-compose.yml:34]: THe `HOST` environment var is still there it's just moved position in the `docker-compose.yml` file. It's now under `env_file` and imported via the `.env.custom` file

File: templates/docker-monolithic/upgrade.sh [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  shift
+done
+
+
+upgrade() {
+    docker-compose down -v
+    time_now=$(date +"%d-%m-%Y-%H:%M:%S")
+    cp -n docker-compose.yml "docker-compose.backup-${time_now}.yml"
+    curl -fLsS https://raw.githubusercontent.com/supportpal/helpdesk-install/master/templates/docker-monolithic/docker-compose.yml -o docker-compose.yml
+    docker-compose up
```
Comment (bytestream) [inline templates/docker-monolithic/upgrade.sh:29]: Is this missing `-d` ?

**Pull request merged**